### ISSUE TITLE
pyproj._datadir does not export expected C variable PJ_CATEGORY_DATUM_ENSEMBLE

### ISSUE THREAD

**Issue by chickymonkeys** [pyproj4/pyproj#791]
Hi, 

I am trying to configure my python environment on my new MacBook Pro with Apple Silicon M1 to run a project I am working on, but I am having troubles importing pyproj. When I try to run `from pyproj import Proj` I obtain the following error in return: 

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/homebrew/Caskroom/miniforge/base/envs/cbkenv/lib/python3.9/site-packages/pyproj/__init__.py", line 49, in <module>
    import pyproj.network
  File "/opt/homebrew/Caskroom/miniforge/base/envs/cbkenv/lib/python3.9/site-packages/pyproj/network.py", line 10, in <module>
    from pyproj._network import (  # noqa: F401
  File "pyproj/_network.pyx", line 1, in init pyproj._network
ImportError: pyproj._datadir does not export expected C variable PJ_CATEGORY_DATUM_ENSEMBLE
```

#### Installation method/steps

 - Installation method (conda, pip wheel, from source, etc...)

I have been trying to install with conda but apparently the package cannot be found, the output of `conda install -c Conda-forge pyproj` is the following:

```
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.

PackagesNotFoundError: The following packages are not available from current channels:

  - pyproj

Current channels:

  - https://conda.anaconda.org/conda-forge/osx-arm64
  - https://conda.anaconda.org/conda-forge/noarch
  - https://repo.anaconda.com/pkgs/main/osx-arm64
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/r/osx-arm64
  - https://repo.anaconda.com/pkgs/r/noarch
```

After that, I have managed to install pyproj using pip in my conda environment.

 - How did you install PROJ? Where is it installed?

I have installed PROJ before installing pyproj with pip, using `conda install PROJ`

#### Environment Information

 - pyproj version you are attempting to install: 3.0.0post1

 - PROJ version (Execute `proj` command and give version here.): `Rel. 8.0.0, March 1st, 2021`
 
 - Python version (`python -c "import sys; print(sys.version.replace('\n', ' '))"`): `3.9.2 | packaged by conda-forge | (default, Feb 21 2021, 05:00:30)  [Clang 11.0.1 ]`
 - Operation System Information (`python -c "import platform; print(platform.platform())"`): `macOS-11.1-arm64-arm-64bit`

**Comment by snowman2** [comment]
Thanks for the report :+1: . This is a known issue (#787) and will be addressed in the next release (#785).

**Comment by snowman2** [comment]
I did some testing and this is actually the fix: https://github.com/pyproj4/pyproj/commit/7044860d6a545d7086e0eab38bbcfcfa08d3cdbd
